src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,118 @@
+
 package core.basesyntax;
 
+import static java.lang.System.arraycopy;
+
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_VALUE = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int RESIZE_INDEX = 2;
+    private Node [] hashMap = new Node[DEFAULT_VALUE];
+    private int currentValue = DEFAULT_VALUE;
+    private int size = 0;
+
+    public class Node<K,V> {
+        private Integer hash;
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        public Node(int hash, K key, V value, Node<K,V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size > currentValue * LOAD_FACTOR) {
+            resize();
+        }
```

Comment:
lets move check `if (size > currentValue * LOAD_FACTOR) {` inside `resize()` method to avoid code duplication

Reply:
We have no any **size** check inside **resize()** method